cmd/node_agent/na/nodeagent.go

```diff
@@ -78,43 +79,98 @@ type platformSpecificRequest interface {
 	IsProperPlatform() bool
 }
 
+// CAGrpcClient is for implementing the GRPC client to talk to CA.
+type CAGrpcClient interface {
+	// Send CSR to the CA and gets the response or error.
+	SendCSR(*string, []grpc.DialOption, *pb.Request) (*pb.Response, error)
+}
+
+// CAGrpcClientImpl is a implementation of GRPC client to talk to CA.
+type CAGrpcClientImpl struct {
+}
+
+// SendCSR sends CSR to CA through GRPC.
+func (c *CAGrpcClientImpl) SendCSR(address *string, options []grpc.DialOption, req *pb.Request) (*pb.Response, error) {
+	conn, err := grpc.Dial(*address, options...)
+	if err != nil {
+		glog.Errorf("Failed to dial %s: %s", *address, err)
```

Comment:
Guess you don't need both return and log the error. Otherwise an error will be logged twice since everything is captured here:https://github.com/istio/auth/pull/243/files#diff-45d9bd931e75161931bda07cf3f93c48R60